src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -350,63 +357,83 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
+        node.parentNode.insertBefore(toInsert, node);
+        return node;
+    };
+
+    /**
+     * Insert separators, indicating the hyphenated positions, into a text node. This will inject the specified
+     * markup as the separators and split the original textnode at the hyphenation points. If there are no places to
+     * inject a hyphenation point, the node will be left unchanged. If the original textnode inclues soft hyphens (i.e.
+     * &shy;) characters, these will be used as a hyphenation point with the original location in the text node
+     * replacded with the `softHyphenPlaceholderMarkup` to allow it to be restored when syllabification is removed.
      *
      * @param {HypherHyphenator} hyphenator - an instance of a Hypher Hyphenator
      * @param {DomNode} node - a DOM node containing text to be hyphenated
      * @param {String} separatorMarkup - the markup to be injected and used to indicate the separators/hyphens
+     * @param {String} softHyphenPlaceholderMarkup - the markup to be injected and used to indicate the original
+     *                                               location of any soft hyphens included in `node`
      */
-    fluid.prefs.enactor.syllabification.hyphenateNode = function (hyphenator, node, separatorMarkup) {
-        if (!hyphenator) {
+    fluid.prefs.enactor.syllabification.hyphenateNode = function (hyphenator, node, separatorMarkup, softHyphenPlaceholderMarkup) {
+        if (!hyphenator || !node.textContent) {
             return;
         }
 
-        // remove \u200B characters added hyphenateText
-        var hyphenated = hyphenator.hyphenateText(node.textContent).replace(/\u200B/gi, "");
+        var hyphenated = hyphenator.hyphenateText(node.textContent);
 
         // split words on soft hyphens
         var segs = hyphenated.split("\u00AD");
+
         // remove the last segment as we only need to place separators in between the parts of the words
         segs.pop();
+
         fluid.each(segs, function (seg) {
-            // Do not attempt to process empty segs, which could lead to an infinite recursion.
+            // If the text content starts with a soft hyphen (\u00AD, &shy;) replace it with an empty placeholder; which
+            // is used to put the soft hyphen back if syllabification is disabled.
             // See: https://issues.fluidproject.org/browse/FLUID-6554
-            if (seg.length) {
-                var separator = $(separatorMarkup)[0]; // converts the separatorMarkup markup string into a Dom Node
-                node = node.splitText(seg.length);
-                node.parentNode.insertBefore(separator, node);
+            if (node.textContent.startsWith("\u00AD")) {
+                // converts the softHyphenPlaceholderMarkup markup string into a Dom Node
+                var placeholder = $(softHyphenPlaceholderMarkup)[0];
+                var newNode = fluid.prefs.enactor.syllabification.insertIntoTextNode(node, placeholder, 1);
```

Comment:
One issue with replacing soft hyphens with something else is hyphenation that indicates the join of the word is gone when the "syllable" setting is turned on:
<img width="286" alt="Screen Shot 2020-10-01 at 4 24 26 PM" src="https://user-images.githubusercontent.com/572877/94859687-eb5b3180-0402-11eb-9ac0-be1918494415.png">

If the soft hyphen is still in place, the hyphenation persists when the "syllable" is on:
<img width="291" alt="Screen Shot 2020-10-01 at 4 24 41 PM" src="https://user-images.githubusercontent.com/572877/94859757-0c238700-0403-11eb-8797-ae1a147e5071.png">

Reply:
Very good remarks, @cindyli!

Reply:
@cindyli I've pushed up some changes which add a soft hyphen to the placeholder pseudo element. Please let me know if this addresses the issue.

Reply:
@cindyli pointed found a bug in the updated implementation if the soft hyphen appears in the last segment, it doesn't get replaced with the placeholder. This is because the algorithm removes the last segment as we don't want to add a separator after it. However, we do need to replace the soft hyphen in all segments.